File: vantix_back/app/schemas/finanzas.py
from pydantic import BaseModel, Field
from datetime import date
from typing import Optional
from decimal import Decimal

class GastoBase(BaseModel):
    id_plan: int
    id_ciente: Optional[int] = None # Mantenemos el nombre de la BD
    fecha_gasto: date
    lugar_origen: Optional[str] = None
    lugar_destino: Optional[str] = None
    institucion_visitada: Optional[str] = None
    motivo_visita: Optional[str] = None
    monto_gastado: Decimal = Field(..., ge=0) # Validamos que sea mayor o igual a 0

class GastoCreate(GastoBase):
    pass

class GastoUpdate(BaseModel):
    id_ciente: Optional[int] = None
    fecha_gasto: Optional[date] = None
    lugar_origen: Optional[str] = None
    lugar_destino: Optional[str] = None
    institucion_visitada: Optional[str] = None
    motivo_visita: Optional[str] = None
    monto_gastado: Optional[Decimal] = None

class GastoResponse(GastoBase):
    id_gasto: int
    class Config:
        from_attributes = True

File: vantix_mobile/src/api/finanzas.ts
import { apiClient } from './client';

export interface GastoData {
    id_plan: number;
    id_ciente?: number;
    fecha_gasto: string;
    lugar_origen: string;
    lugar_destino: string;
    institucion_visitada: string;
    motivo_visita: string;
    monto_gastado: number;
}

export const finanzasService = {
    registrarGasto: async (data: GastoData) => {
        const response = await apiClient.post('/api/v1/finanzas/', data);
        return response.data;
    },
    getGastosByPlan: async (idPlan: number) => {
        const response = await apiClient.get(`/api/v1/finanzas/?id_plan=${idPlan}`);
        return response.data;
    },
};
